packages/blockchain-api/src/transaction/Transaction.ts

```diff
@@ -0,0 +1,138 @@
+import { BigNumber } from 'bignumber.js'
+import { BlockscoutTransferTx } from '../blockscout'
+import { CGLD } from '../currencyConversion/consts'
+import { InputDecoder } from '../helpers/InputDecoder'
+import { FeeType } from '../schema'
+import { Contracts } from '../utils'
+import { TransactionNavigator } from './TransactionNavigator'
+
+export interface Fee {
+  type: FeeType
+  value: BigNumber
+  currencyCode: string
+}
+
+export class Transaction {
+  get transactionHash(): string {
+    return this.blockscoutTx.transactionHash
+  }
+
+  get blockNumber(): string {
+    return new BigNumber(this.blockscoutTx.blockNumber).toFixed()
+  }
+
+  get timestamp(): number {
+    return new Date(this.blockscoutTx.timestamp).getTime()
+  }
+
+  get comment(): string {
+    return this.input.getTransactionComment()
+  }
+
+  get transfers(): TransactionNavigator {
+    return this.transactionNavigator
+  }
+
+  get fees(): Fee[] {
+    return this.transactionFees
+  }
+
+  get input(): InputDecoder {
+    return this.inputDecoder
+  }
+
+  private blockscoutTx: BlockscoutTransferTx
+  private transactionNavigator: TransactionNavigator
+  private transactionFees: Fee[] = []
+  private inputDecoder: InputDecoder
+
+  constructor(
+    blockscoutTx: BlockscoutTransferTx,
+    transactionNavigator: TransactionNavigator,
+    inputDecoder: InputDecoder
+  ) {
+    this.blockscoutTx = blockscoutTx
+    this.transactionNavigator = transactionNavigator
+    this.inputDecoder = inputDecoder
+
+    this.extractFees()
+  }
+
+  addFee(newFee: Fee): void {
+    const feeTypes = this.fees.map((fee) => fee.type)
+    const existingFeeTypeIndex = feeTypes.findIndex((type) => type === newFee.type)
+
+    if (existingFeeTypeIndex !== -1) {
+      const existingFee = this.fees[existingFeeTypeIndex]
+      existingFee.value = existingFee.value.plus(newFee.value)
+    } else {
+      this.fees.push(newFee)
+    }
+  }
+
+  isCeloTransaction(): boolean {
+    return this.blockscoutTx.feeToken !== undefined && this.blockscoutTx.feeToken === CGLD
```

Comment:
Not sure how blockscout treats this, but from the TX side, an empty feeToken means CELO (CGLD) is used to pay for fees, so would suggest to change this to the following to be on the safe side.
```suggestion
    return !this.blockscoutTx.feeToken || this.blockscoutTx.feeToken === CGLD
```

Reply:
Thanks, I accepted the changed and added some tests.